apps/server/src/lib/autoimports/forms/Select/index.svelte

```
@@ -0,0 +1,109 @@
+<script lang="ts">
+  import { Select, Label, Text } from "@latitude-data/svelte";
+  import { setViewParam, useViewParam } from "$lib/stores/viewParams";
+  import { createEventDispatcher } from "svelte";
+  import { type QueryProps, useQuery } from "$lib/stores/queries"
+  import { derived, readable } from "svelte/store"
+  import { type QueryResultState} from "@latitude-data/client"
+
+  type SelectOption = {
+    label: string;
+    value: unknown;
+  } | unknown;
+
+  type $$Props = {
+    param: string;
+    value?: unknown; // Default selected value for the input
+    placeholder?: string;
+    label?: string;
+    description?: string;
+    items?: SelectOption[]; // Manually defined options
+
+    // Adding a query is optional. If not provided, it will just show the child options.
+    query?: string; 
+    labels?: string; // Name of the column to use as labels
+    values?: string; // Name of the column to use as values
+  } & Omit<QueryProps, "query">; // Removed "query" because it's optional here.
+
+  export let param: $$Props["param"];
+  export let value: $$Props["value"] = undefined;
+  export let placeholder: $$Props["placeholder"] = undefined;
+  export let label: $$Props["label"] = undefined;
+  export let description: $$Props["description"] = undefined;
+  export let items: $$Props["items"] = [];
+
+  export let query: $$Props["query"] = undefined;
+  export let inlineParams: $$Props['inlineParams'] = {}
+  export let opts: $$Props['opts'] = {}
+
+  export let labels: $$Props["labels"] = undefined;
+  export let values: $$Props["values"] = undefined;
+
+  let inputStore = useViewParam(param, value);
+
+  const result = query
+    ? useQuery({ query, inlineParams, opts })
+    : readable({ isLoading: false } as QueryResultState, () => {}); // This is a dummy store with no data, for when there's no query.
```

This could be abstracted inside `useQuery`

---

No, I'd rather have `useQuery` fail when used with undefined. This is just an exceptional case, as `query` is optional.

---

Why is good to fail when query is `undefined`? Just to understand

---

`useQuery` is a function that returns reactive content, and it should be used this way. With this in mind, passing `undefined` as the query name does not compute to any actual query, same as when passing a query name that does not exist, so it should fail to warn the user, as it is probably a mistake.

---

But why fail? What's the advantage of failing instead of just doing what you're doing here? also this `useQuery` is implemented in `apps/server` so no other users than the app will use it no?

---

To warn the user, since IMO that's not the desirable usage and may be just a mistake.
Maybe they're using a variable for the query, where it fails when the query does not exist (even if `undefined`), so maybe the variable is not being configured correctly and the user wouldn't notice otherwise.